Index: /trunk/logrotate.c
===================================================================
--- /trunk/logrotate.c	(revision 454)
+++ /trunk/logrotate.c	(revision 455)
@@ -523,4 +523,7 @@
     int i;
     int status;
+	int compressPipe[2];
+	char buff[4092];
+	int error_printed = 0;
 
     message(MESS_DEBUG, "compressing log with: %s\n", log->compress_prog);
@@ -567,4 +570,12 @@
     }
 
+	if (pipe(compressPipe) < 0) {
+		message(MESS_ERROR, "error opening pipe for compress: %s",
+				strerror(errno));
+		close(inFile);
+		close(outFile);
+		return 1;
+	}
+
     if (!FORK_OR_VFORK()) {
 	dup2(inFile, 0);
@@ -572,4 +583,7 @@
 	dup2(outFile, 1);
 	close(outFile);
+	dup2(compressPipe[1], 2);
+	close(compressPipe[0]);
+	close(compressPipe[1]);
 
 	if (switch_user_permanently(log) != 0) {
@@ -581,4 +595,15 @@
     }
 
+    close(compressPipe[1]);
+	while ((i = read(compressPipe[0], buff, sizeof(buff) - 1)) > 0) {
+		if (!error_printed) {
+			error_printed = 1;
+			message(MESS_ERROR, "Compressing program wrote following message "
+					"to stderr when compressing log %s:\n", name);
+		}
+		buff[i] = '\0';
+		fprintf(stderr, "%s", buff);
+	}
+	close(compressPipe[0]);
     wait(&status);
 

