diff -uNr logrotate-3.7.6.orig/examples/logrotate.cron logrotate-3.7.6/examples/logrotate.cron
--- logrotate-3.7.6.orig/examples/logrotate.cron	2003-07-09 19:36:35.000000000 +0000
+++ logrotate-3.7.6/examples/logrotate.cron	2008-01-09 20:24:24.000000000 +0000
@@ -1,8 +1,18 @@
 #!/bin/sh
 
-/usr/sbin/logrotate /etc/logrotate.conf
+MAILTO=""
+ARG=""
+
+if [ -e "/etc/sysconfig/logrotate" ]; then
+	. /etc/sysconfig/logrotate
+fi
+
+/usr/sbin/logrotate ${ARG} /etc/logrotate.conf
 EXITVALUE=$?
 if [ $EXITVALUE != 0 ]; then
     /usr/bin/logger -t logrotate "ALERT exited abnormally with [$EXITVALUE]"
+	if [ "${MAILTO}" != "" ]; then
+		echo "ALERT exited abnormally with [$EXITVALUE]" |  mail -s "logrotate exited abnormally" ${MAILTO}
+	fi
 fi
 exit 0
