--- logrotate-3.7.9/examples/logrotate.cron~	2010-06-28 11:04:56.000000000 +0300
+++ logrotate-3.7.9/examples/logrotate.cron	2010-07-12 10:42:34.890404991 +0300
@@ -1,8 +1,18 @@
 #!/bin/sh
 
-/usr/sbin/logrotate /etc/logrotate.conf
+MAILTO=""
+ARG=""
+
+if [ -e /etc/sysconfig/logrotate ]; then
+	. /etc/sysconfig/logrotate
+fi
+
+/usr/sbin/logrotate $ARG /etc/logrotate.conf >/dev/null 2>&1
 EXITVALUE=$?
 if [ $EXITVALUE != 0 ]; then
-    /usr/bin/logger -t logrotate "ALERT exited abnormally with [$EXITVALUE]"
+	/usr/bin/logger -t logrotate "ALERT exited abnormally with [$EXITVALUE]"
+	if [ -n "$MAILTO" ]; then
+		echo "ALERT exited abnormally with [$EXITVALUE]" |  mail -s "logrotate exited abnormally" $MAILTO
+	fi
 fi
 exit 0
